internal/conf/jwk.go

```diff
@@ -0,0 +1,157 @@
+package conf
+
+import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+type JwtKeysDecoder map[string]JwkInfo
+
+type JwkInfo struct {
+	PublicKey  jwk.Key `json:"public_key"`
+	PrivateKey jwk.Key `json:"private_key"`
+}
+
+// KeyInfo is used to store the initial config of the keys
+// The private key should be in DER format and base64 encoded
+type KeyInfo struct {
+	Type       string `json:"type"`
+	PrivateKey string `json:"private_key"`
+	InUse      bool   `json:"in_use"`
+}
+
+// Decode implements the Decoder interface
+// which transforms the keys stored as der binary strings into jwks
```

Comment:
Hmm but why are any keys stored as DER?

Reply:
Why aren't they just stored as JWK?

Reply:
went with DER initially based on what we discussed earlier but agree that it's much easier if the keys were stored as JWK